src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -1616,9 +1616,13 @@ Expression RegularCondition():
 	| "<" { result = new MinorThan(); }
 	| "=" { result = new EqualsTo(); }
 	| ">=" { result = new GreaterThanEquals(); }
+	| "> =" { result = new GreaterThanEquals("> ="); }
 	| "<=" { result = new MinorThanEquals(); }
+	| "< =" { result = new MinorThanEquals("< ="); }
 	| "<>" { result = new NotEqualsTo(); }
+	| "< >" { result = new NotEqualsTo("< >"); }
```

Comment:
Yes, you are correct that I was handling only a single space.  Ideally, I should handle any whitespace between it – any tips for me on what that would look like in JavaCC?  (I’m having difficulty reconciling the ‘skip’ characters with the insertion of variable whitespace in the token…)

I had originally started by mapping “< =” to just GreaterThanEquals(), but that unfortunately loses information about the ‘style’ of the operator (for example, I can’t go back and slap the hands of the developer for using non-standard relational operators).  I elected to follow the pattern of “NotEqualsTo”, which passed the actual operator text into the constructor.  Is that OK?  Did you have a preference for an alternate way of doing this?

Thanks!


From: Tobias [mailto:notifications@github.com]
Sent: Wednesday, April 27, 2016 4:24 AM
To: JSQLParser/JSqlParser <JSqlParser@noreply.github.com>
Cc: Dave Lindquist <dlindquist@Egistix.com>; Author <author@noreply.github.com>
Subject: Re: [JSQLParser/JSqlParser] Oracle relational operators with spaces & updates to MERGE INTO (#254)


In src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt<https://github.com/JSQLParser/JSqlParser/pull/254#discussion_r61234982>:

>       | "<>" { result = new NotEqualsTo(); }

> +     | "< >" { result = new NotEqualsTo("< >"); }

Isnt this only half the way? Oracle allows, as I learned right now :), multiple whitespaces between < and =. Addionaly wouldnt it be better to map <..= to <=? Is there a case one need to reproduce oracles special space separated syntax?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/JSQLParser/JSqlParser/pull/254/files/6690d717a9d5b771a52a4261480ee692a291ac51#r61234982>